pkg/deploy/lattice/service_manager.go

```diff
@@ -124,7 +124,49 @@ func svcStatusFromCreateSvcResp(resp *CreateSvcResp) ServiceInfo {
 	return svcInfo
 }
 
+func (m *defaultServiceManager) checkAndUpdateTags(ctx context.Context, svc *Service, svcSum *SvcSummary) error {
+	tagsResp, err := m.cloud.Lattice().ListTagsForResourceWithContext(ctx, &vpclattice.ListTagsForResourceInput{
+		ResourceArn: svcSum.Arn,
+	})
+	if err != nil {
+		return err
+	}
+
+	owned, err := m.cloud.CheckAndAcquireOwnershipFromTags(ctx, *svcSum.Arn, tagsResp.Tags)
+	if err != nil {
+		return err
+	}
+	if !owned {
+		return services.NewConflictError("service", svc.Spec.RouteNamespace+"/"+svc.Spec.RouteName,
+			fmt.Sprintf("Found existing resource not owned by controller: %s", *svcSum.Arn))
+	}
+
+	tagFields := model.ServiceTagFieldsFromTags(tagsResp.Tags)
+	switch {
+	case tagFields.RouteName == "" && tagFields.RouteNamespace == "":
+		// backwards compatibility: If the service has no identification tags, consider this controller has
+		// correct information and add tags
+		_, err = m.cloud.Lattice().TagResourceWithContext(ctx, &vpclattice.TagResourceInput{
+			ResourceArn: svcSum.Arn,
+			Tags:        svc.Spec.ToTags(),
+		})
+		return err
+	case tagFields != svc.Spec.ServiceTagFields:
+		// Considering these scenarios:
+		// - two services with same namespace-name but different routeType
+		// - two service with conflict edge case such as my-namespace/service & my/namespace-service
+		return services.NewConflictError("service", svc.Spec.RouteName+svc.Spec.RouteNamespace,
+			fmt.Sprintf("Found existing resource with conflicting service name: %s", *svcSum.Arn))
+	}
+	return nil
+}
+
 func (m *defaultServiceManager) updateServiceAndAssociations(ctx context.Context, svc *Service, svcSum *SvcSummary) (ServiceInfo, error) {
+	err := m.checkAndUpdateTags(ctx, svc, svcSum)
+	if err != nil {
+		return ServiceInfo{}, err
+	}
```

Comment:
It's probably ok for near future to imply that ownership and tags are related. And operate on them together to avoid redundant getTags calls. But conceptually ownership should be abstracted. For example this function tells that we do something with Tags, but underneath we manage ownership.